redfish/provider/resource_redfish_manager_reset.go

```diff
@@ -0,0 +1,234 @@
+				stringvalidator.OneOf(
+					string(redfish.GracefulRestartResetType),
+				),
+			},
+		},
+	}
+}
+
+// Schema defines the schema for the resource.
+func (*managerResetResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Resource for managing manager reset.",
+
+		Attributes: ManagerResetSchema(),
+		Blocks:     RedfishServerResourceBlockMap(),
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *managerResetResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Trace(ctx, "resource_manager_reset create : Started")
+	// Get Plan Data
+	var plan models.RedfishManagerReset
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Lock the mutex to avoid race conditions with other resources
+	redfishMutexKV.Lock(plan.RedfishServer[0].Endpoint.ValueString())
+	defer redfishMutexKV.Unlock(plan.RedfishServer[0].Endpoint.ValueString())
+
+	resetType := plan.ResetType.ValueString()
+	managerID := plan.ManagerID.ValueString()
+
+	// Get manager
+	manager, err := getManager(r, plan, managerID)
+	if err != nil {
+		resp.Diagnostics.AddError("Error while retrieving manager from redfish API", err.Error())
+		return
+	}
+
+	// Perform manager reset
+	err = manager.Reset(redfish.ResetType(resetType))
+	if err != nil {
+		resp.Diagnostics.AddError("Error resetting manager", err.Error())
+		return
+	}
+
+	plan.Id = types.StringValue(manager.ODataID)
+
+	// time to allow rebooting iDRAC
+	time.Sleep(300 * time.Second)
+
+	tflog.Trace(ctx, "resource_manager_reset create: updating state finished, saving ...")
+	// Save into State
+	diags = resp.State.Set(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	tflog.Trace(ctx, "resource_manager_reset create: finish")
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *managerResetResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	tflog.Trace(ctx, "resource_manager_reset read: started")
+	var state models.RedfishManagerReset
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	managerID := state.ManagerID.ValueString()
+
+	// Get manager
+	manager, err := getManager(r, state, managerID)
+	if err != nil {
+		resp.Diagnostics.AddError("Error while retrieving manager from redfish API", err.Error())
+		return
+	}
+
+	state.Id = types.StringValue(manager.ODataID)
+
+	// Save into State
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	tflog.Trace(ctx, "resource_manager_reset read: finished")
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *managerResetResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
So the update function wd be called and the iDRAC wd be reset even when the user updates just the reset type (unless explicitly disabled by the user using the lifecycle controllers). Is this the functionality our users are expecting?

Reply:
we are not performing iDRAC reset on update here but I think we should perform reset either if manager id or reset type is updated. Because if user is updating it, he is expecting it to reset now.

Reply:
will make those changes.

Reply:
Okay